Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPMediaSink.cpp

```diff
@@ -0,0 +1,211 @@
+        if (dwStreamSinkIdentifier > m_spStreamSinks.size())
+        {
+            return MF_E_INVALIDSTREAMNUMBER;
+        }
+        else
+        {
+            m_spStreamSinks[dwStreamSinkIdentifier].as<IMFStreamSink>().copy_to(ppStreamSink);
+            return S_OK;
+        }
+    }
+
+    STDMETHODIMP SetPresentationClock(
+        /* [in] */ __RPC__in_opt IMFPresentationClock* pPresentationClock) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (m_spClock)
+        {
+            m_spClock->RemoveClockStateSink(this);
+            m_spClock = nullptr;
+        }
+        m_spClock.copy_from(pPresentationClock);
+        return m_spClock->AddClockStateSink(this);
+
+    }
+
+    STDMETHODIMP GetPresentationClock(
+        /* [out] */ __RPC__deref_out_opt IMFPresentationClock** ppPresentationClock) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (!ppPresentationClock) return E_POINTER;
+        m_spClock.copy_to(ppPresentationClock);
+        return S_OK;
+    }
+
+    STDMETHODIMP Shutdown(void) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        HRESULT hr = S_OK;
+        m_bIsShutdown = true;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Shutdown();
+            hr = FAILED(hr1)? hr1: hr;
+        }
+        return hr;
+    }
+
+    //IMediaExtension
+    STDMETHODIMP SetProperties(
+        winrt::Windows::Foundation::Collections::IPropertySet const& configuration) noexcept
+    {
+        return S_OK;
+    }
+
+    // IMFClockStateSink methods
+    STDMETHODIMP OnClockStart(MFTIME hnsSystemTime, LONGLONG llClockStartOffset) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Start(hnsSystemTime, llClockStartOffset);
+            hr = FAILED(hr1) ? hr1 : hr;
+        }
+        return hr;
+    }
+    STDMETHODIMP OnClockStop(MFTIME hnsSystemTime) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Stop(hnsSystemTime);
+            hr = FAILED(hr1) ? hr1 : hr;
+        }
+        return hr;
+    }
+    STDMETHODIMP OnClockPause(MFTIME hnsSystemTime) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Pause(hnsSystemTime);
+            hr = FAILED(hr1) ? hr1 : hr;
+        }
+        return hr;
+    }
+    STDMETHODIMP OnClockRestart(MFTIME hnsSystemTime) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Start(hnsSystemTime, 0);
+            hr = FAILED(hr1) ? hr1 : hr;
+        }
+        return hr;
+    }
+    STDMETHODIMP OnClockSetRate(MFTIME hnsSystemTime, float flRate) noexcept
+    {
+        return S_OK;
+    }
+};
+
+RTPMEDIASTREAMER_API STDMETHODIMP CreateRTPMediaSink(IMFMediaType** apMediaTypes, DWORD dwMediaTypeCount, IMFMediaSink** ppMediaSink)

```

Comment:
should this have noexcept

Reply:
Do I need a noexcept even when i am using an exception boundary? The exception boundary is using winrt::to_hresult which takes care of calling std::terminate on unhandled exceptions, which is what the compiler does for noexcept.

---
In reply to: [491168094](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491168094) [](ancestors = 491168094)